### PR TITLE
Update Core to use Azure Json

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Adds `azure-json` as a dependency for `azure-core` and replaced reflective usage with direct usage of APIs in `azure-json`.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by joshfree** [comment]
image

File: sdk/core/azure-core-experimental/pom.xml [hunk 1]
```diff
@@ -72,16 +72,15 @@
   <dependencies>
     <dependency>
       <groupId>com.azure</groupId>
-      <artifactId>azure-core</artifactId>
-      <version>1.38.0-beta.1</version> <!-- {x-version-update;com.azure:azure-core;current} -->
+      <artifactId>azure-json</artifactId>
```
Comment (srnagar) [inline sdk/core/azure-core-experimental/pom.xml:75]: We can drop this dependency as azure-core brings this transitively.

File: sdk/core/azure-core-experimental/src/main/java/module-info.java [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the MIT License.
 
 module com.azure.core.experimental {
+    requires transitive com.azure.json;
```
Comment (srnagar) [inline sdk/core/azure-core-experimental/src/main/java/module-info.java:5]: Is this needed? Azure core's transitive `requires` will be sufficient.

**Review by srnagar — APPROVED** [review]
LGTM!

File: sdk/core/azure-core-experimental/src/main/java/module-info.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/core/azure-core-experimental/src/main/java/module-info.java:5]: Good catch, at this time I think the only usage is internal, so it won't need to be transitive.

File: sdk/core/azure-core-experimental/pom.xml (hunk 1 above)
Comment (alzimmermsft) [reply sdk/core/azure-core-experimental/pom.xml:75]: I prefer having libraries directly including dependencies used directly. Helps with transitive dependencies being overridden based on Maven dependency resolution.

**Pull request merged**